loader/db.go

```diff
@@ -273,3 +291,215 @@ func isMySQLError(err error, code uint16) bool {
+	dataNow := 0    // Record the start of each match
+	nowPos := 0     // Record the location of the current data, and compare whether the data is equal to dupData only when it is equal to pos
+	isSymbol := 0   // Records whether the current data carries ', Used to record data locations and compare data, 0 for stateless, 1 for currently traversing string data, 2 for currently traversing non-string data
+	isFind := false // Use a variable to record whether duplicate values have been found
+	for i := 0; i < len(target); i++ {
+		switch target[i] {
+		case '(':
+			if dataIn == false {
+				trxStart = i             // Left edge position of deleted data
+				if target[i+1] != '\'' { // For non-string data after (
+					nowPos++
+					dataNow = i
+					isSymbol = 2
+				}
+			}
+		case '\\': // When an escape character is encountered, need to skip the next character to avoid the effect of characters such as ' "
+			i++
+			continue
+		case '\'':
+			if dataIn == false {
+				dataIn = true
+				isSymbol = 1
+				nowPos++
+				dataNow = i
+			} else {
+				dataIn = false
+				isSymbol = 0
+				if nowPos == pos { // Find a string data
+					if target[dataNow+1:i] == dupData {
+						isFind = true
+					}
+				}
+			}
+		case ',':
+			if isSymbol == 0 && dataIn == false && target[i-1] != ')' && target[i+1] != '\'' { // No "," can appear before the beginning of non-string data, and cannot between the string data, but between the "(" and ")", and the last bit cannot be a "'"
+				nowPos++
+				dataNow = i
+				isSymbol = 2
+			} else if isSymbol == 2 {
+				isSymbol = 0
+				if nowPos == pos { // Find a non-string data
+					if target[dataNow+1:i] == dupData {
+						isFind = true
+					}
+				}
+				if target[i+1] != '\'' { // The "," may be the beginning of another data at the same time as the end of the data
+					nowPos++
+					dataNow = i
+					isSymbol = 2
+				}
+			}
+		case ')':
+			if dataIn == false {
+				if isSymbol == 2 { // For non-string data before )
+					isSymbol = 0
+					if nowPos == pos { // Find a non-string data
+						if target[dataNow+1:i] == dupData {
+							isFind = true
+						}
+					}
+				}
+				nowPos = 0
+				if isFind == true {
+					if target[i+1] == ';' { // If the next digit is ";", delete the "," before the data
+						tmpSymbol := 2 // Find a "," or "S" that before the duplicate data pair
+						for target[trxStart-tmpSymbol] != ',' && target[trxStart-tmpSymbol] != 'S' {
+							tmpSymbol++
+						}
+						delRes = target[:trxStart-tmpSymbol] + target[i+1:]
+					} else {
+						delRes = target[:trxStart] + target[i+2:]
+					}
+					dupDataPair = target[trxStart : i+1]
+					return delRes, dupDataPair, nil
+				}
+			}
+		}
+	}
+	return delRes, "", errors.New("Can't find dup Error")
+}
+
+func errorLog(dupDataSQL, errorTable, errorData, errorKey, errorPos string) error {
+	path := "./dupDataFile.txt"
```

Comment:
how to use this file?

Reply:
This file will appear in the catalog of each dm-worker, and we can use the catalog to confirm the file comes from which mysql instance, and file records the conflict value, conflict key, the column location of the conflict value, and an insert SQL
*********************
Key: b
Data: repqwd
Pos: 2
INSERT INTO `t3` VALUES('123456','repqwd','66668888',2);

Reply:
can we just save these information into log?